### PR TITLE
The README Manifesto

### PR DESCRIPTION
**Pull request by JonathanWbn**
**tl;dr**

I suggest that the best way to enable developers at MOIA to configure typing, linting, and formatting in their projects is with documentation instead of shared configs. The documentation would be [this README](https://github.com/moia-oss/eslint-prettier-typescript-config/blob/reset/README.md).

---

### Why a manifesto

Many typescript services at MOIA use the shared ESLint configs of this project. ESLint has introduced a new config format that everyone has to use. The configs of this project are not compatible with the new format, so they have to be migrated to unblock all these MOIA services from updating their ESLint dependencies.

### The obvious solution I want to reject

I have tried updating the existing shared ESLint configs to the new format. I have found this to be painfully difficult. I have found the source for this pain to be the complexity of these shared-config setups. I now believe that these shared configs create more problems than they solve. So instead of fixing them, I want to remove them.

 Having migrated the complex [backoffice-frontend ESLint config](https://github.com/moia-dev/backoffice-frontend/pull/8327) I can report that I would rather _not_ use this shared config (even if it was compatible with the new format). It's another layer of complexity I want to avoid.

### What then

There are two main goals I see for shared configs:
1. enable developers to quickly set up and configure their project
2. ensure consistency across MOIA to allow developers to switch between projects with little friction

I think this project once achieved both of these goals. I think it currently achieves none of them. That's because it's really difficult to maintain an opinionated set of configs. And today these configs are outdated and partly broken.

I want to argue that we should just give up on the second goal of ensuring consistency. I don't think we want to be consistent, but instead, we want to stay up-to-date and follow best practices. A good way to achieve this is to regularly update the dependencies (_stay up-to-date_) and use the tools defaults (_follow best practices_). I think we, as MOIA, don't want to have an opinion on JavaScript formatting. We just want to do what everyone else does.

But I do believe in the first goal of enabling developers to get started quickly. And I think this is best achieved with a slightly opinionated but basic getting-started documentation. And that is what remains with this PR. [Link to the remaining README](https://github.com/moia-oss/eslint-prettier-typescript-config/blob/reset/README.md)

---

**But what about all the projects that still use the configs?**
They can continue using them. When they run into issues (e.g., while upgrading ESLint), they will check back here and find a readme that should help them get unstuck.

### CODE REVIEW

**Comment by jan-xyz** [comment]
>  use the tools defaults (follow best practices)

A virtue that is all to often forgotten ❤️
